tooling/verifier-server/README.md

```diff
@@ -0,0 +1,71 @@
+# ProveKit Verifier Server
+
+HTTP server combining Rust (API) + Go (verifier binary) for WHIR-based proof verification.
+
+## Quick Start
+
+```bash
+cd tooling/verifier-server
+docker-compose up --build
+```
+
+Server runs at `http://localhost:3000`
+
+## API
+
+### Health Check
+```bash
+curl http://localhost:3000/health
+```
+
+### Verify Proof
+```bash
+curl -X POST http://localhost:3000/verify \
+  -H "Content-Type: application/json" \
+  -d '{
+    "npsUrl": "https://example.com/scheme.nps",
+    "r1csUrl": "https://example.com/r1cs.json", 
+    "pkUrl": "https://example.com/proving_key.bin", (optional)
+    "vkUrl": "https://example.com/verification_key.bin", (optional)
+    "np": { /* NoirProof JSON */ },
+  }'
+```
+
+**Response:**
+```json
+{
+  "isValid": true,
+  "result": {
+    "status": "valid",
+    "verificationTimeMs": 1500
+  },
+  "metadata": {
+    "serverVersion": "0.1.0",
+    "requestId": "unique-request-id"
+  }
+}
+```
+
+## Build Options
+
+```bash
+# Docker (recommended)
+./build.sh
+docker-compose up --build
+
+# Local development
+cargo run --bin verifier-server
+```
+
+## Environment Variables
```

Comment:
update this to include `VERIFIER_SEMAPHORE_LIMIT`, please check if the README needs any more updation.